src/helpers/RangeFinder.js

```diff
@@ -7,6 +7,17 @@ function search(element, phrase) {
   }));
 }
 
-export function findRanges(node, patternsToFind) {
-  return patternsToFind.flatMap((pattern) => search(node, pattern));
+export function findRanges(target, patternsToFind) {
```

Comment:
`findRanges` doesn't seems like an accurate name.  I know it's been in the code for quite a while, but your refactorings have helped me realize that it's not just finding ranges.  It's returning a list of all the warnings for a particular email plus metadata like range.  This function contains the core business logic of Just Not Sorry.  We need a name that reflects its purpose.  Maybe `calculateWarnings`?